pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ServerSyncJob.kt

```diff
@@ -1,35 +1,44 @@
 package com.pusher.pushnotifications.internal
 
 import com.pusher.pushnotifications.api.DeviceMetadata
-import com.squareup.moshi.adapters.PolymorphicJsonAdapterFactory
+import com.squareup.moshi.JsonClass
+import dev.zacsweers.moshisealed.annotations.TypeLabel
 import java.io.Serializable
 
-sealed class ServerSyncJob: Serializable {
-
-    companion object {
-        // the following is required for the Persistent Job Queue to read and write data correctly.
-        // if you add another ServerSyncJob, you must also add it below here too!
-        val jsonAdapterFactory: PolymorphicJsonAdapterFactory<ServerSyncJob>
-                = PolymorphicJsonAdapterFactory.of(ServerSyncJob::class.java, "ServerSyncJob")
-                .withSubtype(StartJob::class.java, "StartJob")
-                .withSubtype(RefreshTokenJob::class.java, "RefreshTokenJob")
-                .withSubtype(SubscribeJob::class.java, "SubscribeJob")
-                .withSubtype(UnsubscribeJob::class.java, "UnsubscribeJob")
-                .withSubtype(SetSubscriptionsJob::class.java, "SetSubscriptionsJob")
-                .withSubtype(ApplicationStartJob::class.java, "ApplicationStartJob")
-                .withSubtype(SetUserIdJob::class.java, "SetUserIdJob")
-                .withSubtype(StopJob::class.java, "StopJob")
-    }
-}
+@JsonClass(generateAdapter = true, generator = "sealed:ServerSyncJob")
+sealed class ServerSyncJob: Serializable { }
 
 // If you add new fields to any of these data classes, make sure they have
 // default values otherwise they can be null when reading the old stored
 // JSON from disk.
```

Comment:
That comment have been no longer valid already before the changes in this PR but now it's a good opportunity to delete it.

Reply:
@daniellevass can you move some of your PR comment you did [here](https://github.com/pusher/push-notifications-android/pull/101#issuecomment-567611532) to this comment.

Reply:
Have updated the comment to point people to the tests for the single source of truth! I updated the tests to more explicitly detail what results are expected and added comments there if data is expected not to be returned e.g. if a field is missing. Does that make sense?